config-scalecube-configuration-service/src/main/java/io/scalecube/config/service/ScalecubeConfigurationServiceConfigSource.java

```diff
@@ -0,0 +1,124 @@
+import io.scalecube.config.ConfigSourceNotAvailableException;
+import io.scalecube.config.source.ConfigSource;
+import io.scalecube.config.source.LoadedConfigProperty;
+import io.scalecube.configuration.api.ConfigurationService;
+import io.scalecube.configuration.api.EntriesRequest;
+import io.scalecube.configuration.api.FetchResponse;
+import io.scalecube.services.gateway.clientsdk.ClientSettings.Builder;
+import io.scalecube.services.transport.jackson.JacksonCodec;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.netty.http.HttpResources;
+
+public class ScalecubeConfigurationServiceConfigSource implements ConfigSource {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(ScalecubeConfigurationServiceConfigSource.class);
+
+  ConfigurationService service;
+
+  private EntriesRequest requestEntries;
+
+  private final Parsing parsing = new Parsing();
+
+  private static URL defaultUrl() {
+    try {
+      return new URL("https", "configuration-service-http.genesis.om2.com", 443, "/");
+    } catch (MalformedURLException ignoredException) {
+      return null;
+    }
+  }
+
+  /**
+   * Create a configuration source that connects to the production environment of scalecube
+   * configuration service.
+   *
+   * @param token the API token
+   * @param repository the name of the repository
+   */
+  public ScalecubeConfigurationServiceConfigSource(String token, String repository) {
+    this(token, repository, defaultUrl());
+  }
+
+  /**
+   * Create a configuration source that connects to the production environment of scalecube
+   * configuration service.
+   *
+   * @param token the API token
+   * @param repository the name of the repository
+   * @param service the URL of this service
+   */
+  public ScalecubeConfigurationServiceConfigSource(String token, String repository, URL service) {
+    Builder builder =
+        builder()
+            .host(service.getHost())
+            .port(service.getPort())
+            .contentType(JacksonCodec.CONTENT_TYPE)
+            .loopResources(HttpResources.get());
+    if ("https".equals(service.getProtocol()) || "wss".equals(service.getProtocol())) {
+      builder = builder.secure();
+    }
+    this.service = http(builder.build()).forService(ConfigurationService.class);
+    requestEntries = new EntriesRequest(token, repository);
+  }
+
+  /**
+   * Create a configuration source that connects to the production environment of scalecube
+   * configuration service.
+   *
+   * @param token the API token
+   * @param repository the name of the repository
+   * @param service the actual configuration service
+   */
+  public ScalecubeConfigurationServiceConfigSource(
+      String token, String repository, ConfigurationService service) {
+    this.service = service;
+    requestEntries = new EntriesRequest(token, repository);
```

Comment:
Maybe just put `EntriesRequest` as a parameter instead of having two: `token`, and `repository`.  This you would decouple from signature constructor of `EntriesRequest` .

Reply:
It makes no sense to do this. it's like exposing the way this config-source is working (and hereafter locking the implementation)